x-pack/plugins/siem/public/common/components/exceptions/viewer/exception_entries.test.tsx

```diff
@@ -0,0 +1,149 @@
+          entries={[getFormattedEntryMock(), getFormattedEntryMock()]}
+          handleDelete={jest.fn()}
+          handleEdit={jest.fn()}
+        />
+      </ThemeProvider>
+    );
+
+    expect(wrapper.find('AndOrBadge[data-test-subj="exceptionsViewerAndBadge"]')).toHaveLength(1);
+  });
+
+  test('it invokes "handlEdit" when edit button clicked', () => {
+    const mockHandleEdit = jest.fn();
+    const wrapper = mount(
+      <ThemeProvider theme={() => ({ eui: euiLightVars, darkMode: false })}>
+        <ExceptionEntries
+          entries={[getFormattedEntryMock()]}
+          handleDelete={jest.fn()}
+          handleEdit={mockHandleEdit}
+        />
+      </ThemeProvider>
+    );
+    const editBtn = wrapper.find('[data-test-subj="exceptionsViewerEditBtn"] button').at(0);
+    editBtn.simulate('click');
+
+    expect(mockHandleEdit).toHaveBeenCalledTimes(1);
+  });
+
+  test('it invokes "handleDelete" when delete button clicked', () => {
+    const mockHandleDelete = jest.fn();
+    const wrapper = mount(
+      <ThemeProvider theme={() => ({ eui: euiLightVars, darkMode: false })}>
+        <ExceptionEntries
+          entries={[getFormattedEntryMock()]}
+          handleDelete={mockHandleDelete}
+          handleEdit={jest.fn()}
+        />
+      </ThemeProvider>
+    );
+    const deleteBtn = wrapper.find('[data-test-subj="exceptionsViewerDeleteBtn"] button').at(0);
+    deleteBtn.simulate('click');
+
+    expect(mockHandleDelete).toHaveBeenCalledTimes(1);
+  });
+
+  test('it renders nested entry', () => {
+    const parentEntry = getFormattedEntryMock();
+    parentEntry.operator = null;
+    parentEntry.value = null;
+
+    const wrapper = mount(
+      <ThemeProvider theme={() => ({ eui: euiLightVars, darkMode: false })}>
+        <ExceptionEntries
+          entries={[parentEntry, getFormattedEntryMock(true)]}
+          handleDelete={jest.fn()}
+          handleEdit={jest.fn()}
+        />
+      </ThemeProvider>
+    );
+
+    const parentField = wrapper
+      .find('EuiTableRowCell[data-test-subj="exceptionFieldNameCell"] .euiTableCellContent')
+      .at(0);
+    const parentOperator = wrapper
+      .find('EuiTableRowCell[data-test-subj="exceptionFieldOperatorCell"] .euiTableCellContent')
+      .at(0);
+    const parentValue = wrapper
+      .find('EuiTableRowCell[data-test-subj="exceptionFieldValueCell"] .euiTableCellContent')
+      .at(0);
+
+    const nestedField = wrapper
+      .find('EuiTableRowCell[data-test-subj="exceptionFieldNameCell"] .euiTableCellContent')
+      .at(1);
+    const nestedOperator = wrapper
+      .find('EuiTableRowCell[data-test-subj="exceptionFieldOperatorCell"] .euiTableCellContent')
+      .at(1);
+    const nestedValue = wrapper
+      .find('EuiTableRowCell[data-test-subj="exceptionFieldValueCell"] .euiTableCellContent')
+      .at(1);
+
+    expect(parentField.text()).toEqual('host.name');
+    expect(parentOperator.text()).toEqual('—');
+    expect(parentValue.text()).toEqual('—');
```

Comment:
For tests of empty, rather than hard code them you will be better off using

```ts
toEqual(getEmptyValue())
```

So that if the aesthetics of double dash/hyphen ever changes we won't have to update tests but rather just update that one function in one spot and all tests still pass.

Reply:
Good call. Updated - thanks!